backend/pkg/factory/kafka/metrics.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2025 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package kafka
+
+import (
+	"sync"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+// FactoryMetrics holds all Kafka client factory metrics
+type FactoryMetrics struct {
+	ActiveClientsGauge    prometheus.Gauge
+	ClientsCreatedCounter prometheus.Counter
+	ClientsClosedCounter  prometheus.Counter
+}
+
+var (
+	factoryMetricsRegistry = make(map[string]*FactoryMetrics)
+	factoryMetricsMu       sync.RWMutex
+)
+
+// NewFactoryMetrics creates and registers factory metrics for client tracking
+func NewFactoryMetrics(metricsNamespace string, clientType string, registry prometheus.Registerer) *FactoryMetrics {
```

Comment:
NIT: I was wondering what the clientType string was for, maybe that would clarify it better.

```suggestion
// NewFactoryMetrics creates and registers factory metrics for client tracking
// The provided factoryType is added a const label to all emitted metrics.
func NewFactoryMetrics(metricsNamespace string, factoryType string, registry prometheus.Registerer) *FactoryMetrics {
```